src/sentry/data_secrecy/api/waive_data_secrecy.py

```diff
@@ -0,0 +1,146 @@
+    access_end = serializers.DateTimeField()
+
+    def validate(self, data):
+        access_start = data.get("access_start")
+        access_end = data.get("access_end")
+
+        if access_start >= access_end:
+            raise serializers.ValidationError(
+                "Invalid timestamp (access_start must be before access_end)."
+            )
+        if access_end <= timezone.now():
+            raise serializers.ValidationError(
+                "Invalid timestamp (access_end must be in the future)."
+            )
+
+        return data
+
+    def serialize(
+        self,
+        obj: DataSecrecyWaiver,
+        attrs: Sequence[Any],
+        user: User,
+        **kwargs: Any,
+    ) -> serializers.Mapping[str, Any]:
+        return {
+            "access_start": obj.access_start.isoformat(),
+            "access_end": obj.access_end.isoformat(),
+            "zendesk_tickets": list(obj.zendesk_tickets),
+        }
+
+
+@region_silo_endpoint
+class WaiveDataSecrecyEndpoint(OrganizationEndpoint):
+    permission_classes = [WaiveDataSecrecyPermission]
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+        "PUT": ApiPublishStatus.PRIVATE,
+        "DELETE": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.ENTERPRISE
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        """
+        Returns the data secrecy waiver for an organization if it exists.
+        """
+        try:
+            ds = get_object_or_404(DataSecrecyWaiver, organization=organization)
+            return Response(
+                serialize(ds, request.user, DataSecrecyWaiverSerializer()),
+                status=status.HTTP_200_OK,
+            )
+        except Http404:
+            return Response(
+                {"detail": "No data secrecy waiver in place."}, status=status.HTTP_404_NOT_FOUND
+            )
+
+    def put(self, request: Request, organization: Organization):
+        """
+        Update or create an entry for an organization with data secrecy waived until a specified timestamp
+
+        :param access_start: The timestamp at which data secrecy is waived
+        :param access_end: The timestamp at which data secrecy is reinstated
+        """
+        serializer = DataSecrecyWaiverSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        access_start = serializer.validated_data["access_start"]
+        access_end = serializer.validated_data["access_end"]
+
+        ds, _ = DataSecrecyWaiver.objects.update_or_create(
+            organization=organization,
+            defaults={
+                "access_start": access_start,
+                "access_end": access_end,
+            },
+        )
+
+        self.create_audit_entry(
+            request=request,
+            organization=organization,
+            event=audit_log.get_event_id("DATA_SECRECY_WAIVED"),
```

Comment:
do we care that this audit log entry will be emitted every time somebody adjusts the access period?

Reply:
not sure? i think that waiving shouldn't happen that often and customers should know who is adjusting this.